### ISSUE TITLE
Questions on features

### ISSUE THREAD

**Issue by gergelyrakoczi** [rwthanalytics/moodle-local_learning_analytics#18]
Hi,
thank you for this very promising plugin for Moodle. We love your approach on using logdata and are considering to switch from our current tool to yours. What we still wondering about is ...

- Is there a way to filter usage statictsics for a specific activity/resource for a specific time frame (e.g. March 17th to April 10th file XYZ)?
- Is it possible to write a Moodle message (anonymously) to those students who were active or inactive at a specific activity/resource?
- Is it possible to see usage statistics not only for view action but also update/submit etc.?
- Do you support / will consider group specific usage staticstics?

Thank you for your help in advance and keep up the great work!

Best regards / herzliche Grüße,
Gergely Rakoczi (TU Wien)

**Comment by thomasdondorf** [comment]
Hi Gergely,

that's nice to hear. The TU Wien would be the 10th university (as far as I'm aware) to adopt the system in German speaking countries ;)

I'll address your points below:

- It's currently not possible to filter by time. But this has been requested multiple times in the past and is still on the TODO list. But the time resources for this plugin are currently rather limited. So don't expect this to be implemented soon.
- It's not possible to address individual students as no data of individual students is logged. All stored data is stored on the course level without a way to trace it back to individual learners. That's a big privacy feature but of course it comes at a cost...
- This is not implemented (and there are no plans to) but it's technically possible.
- See my previous point regarding individual students. It's not possible to group the logged data for specific students.

Best,
Thomas

**Comment by gergelyrakoczi** [comment]
Hi Thomas,

thank you for your fast response and I'm happy to hear that some of our requests are technically possible and are noted. We will check internally how to proceeed and yes, it would be great to be the 10th German-speaking University. :)

Best regards,
Gergely

**Comment by kaspot** [comment]
Hi Thomas, 
as a follow-up to Gergely's messages (also TU Wien) I would like to propose something ;-) 
We (TU Wien) would love to use your plugin but there are a few features, as Gergely already mentioned, which are still missing and very often requested and already used at TU Wien. 
So what we would like to propose is that we would tackle those issues, implement those features, create pull requests, and would like to bring those upstream. Now, this makes only sense if you would be willing to integrate our newly developed features into your plugin. 

The most important features that we would like to see in your plugin are: 
- filter usage statistics for a specific time frame (e.g. March 17th to April 10th file XYZ)
- see usage statistics not only for view action but also update/submit/started etc.
- filter usage statistics for a **specific** activity/resource for a specific time frame (e.g. March 17th to April 10th file XYZ)
- a nicer and more readable description of the activities in the graph (if the names are too long it gets really ... well ;-))

Very low priority: 
- group filter

I would appreciate it if you could let me know if that would be a possibility for you - would love to see some cooperation here ;-)

Thanks
Kasia

**Comment by thomasdondorf** [comment]
Hi @kaspot

Sounds great, waiting for your pull request! 😊 If you need help, let me know.

Please keep in mind that everything should also work in big courses with a few thousand of enrolled users.

Best
Thomas

**Comment by kaspot** [comment]
Hey Thomas, 
Cool - that means that you would be willing to add all of our enhancements and developments to the official code - that's pretty cool - so we will start working on them hopefully this year. 
Best Kasia

**Comment by thomasdondorf** [comment]
Yes, especially the "time frame" has been requested from multiple lecturers on our side, too...
But please keep in mind that the user experience should not suffer. The lecturers should still be able to use the page without studying PDF instructions before they are able to use the tool...

I'm not sure how exactly you are planning on developing this, but if you want you can open one issue or pull request for each feature you mentioned to tackle them one after another. I can even add you to the repo as developer so you can create your own branch(es) to work in if you are interested. This might be easier to integrate in multiple smaller parts compared to "one big branch". And if you need help you can ping me.

Best, Thomas
